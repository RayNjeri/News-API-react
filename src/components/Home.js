import React from 'react';
import Header from './Header';
import Content from './Content';
import Login from './Login';
import Footer from './Footer';

export default class Home extends React.Component {
  render() {
    return (
      <div className="home">
        <div className="headers"><Header /></div>
        <div className="login"><Login /></div>
        <div className="content"><Content /></div>
        <Footer />
      </div >
    );
  }
}
