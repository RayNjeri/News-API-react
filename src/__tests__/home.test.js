import React from 'react';
import { mount, shallow } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';
import Home from '../components/Home';

describe('Home Component', () => {
    it('Should render a home page', () => {
        const homes = shallow(<Home />);
        const tree = shallowToJSON(homes);
        expect(tree).toMatchSnapshot();
    });
    it('should render header div class', () => {
        const header = shallow(<Home />);
        expect(header.find('Header')).toEqual.defined;
    });

});