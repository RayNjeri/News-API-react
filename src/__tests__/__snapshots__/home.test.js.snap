// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Should render a home page 1`] = `
<div
  className="home"
>
  <div
    className="headers"
  >
    <Header />
  </div>
  <div
    className="login"
  >
    <Login />
  </div>
  <div
    className="content"
  >
    <Content />
  </div>
  <Footer />
</div>
`;
